napari/layers/intensity_mixin.py

```diff
@@ -40,15 +45,16 @@ def reset_contrast_limits(self: 'Image', mode=None):
         mode = mode or self._auto_contrast_source
         self.contrast_limits = self._calc_data_range(mode)
 
-    def reset_contrast_limits_range(self):
-        """Scale contrast limits range to data type.
-
-        Currently, this only does something if the data type is an integer...
-        otherwise it's unclear what the full range should be.
+    def reset_contrast_limits_range(self, mode=None):
+        """Scale contrast limits range to data type if dtype is an integer,
+        or use the current maximum data range otherwise.
         """
-        if np.issubdtype(self.dtype, np.integer):
+        if np.issubdtype(self.dtype.type, np.integer):
```

Comment:
Not that this change was necessary for `tensorstore` dtype to not fail. The bug was already there, we just don't have any test that ran `reset_contrast_limits_range` with `tensorstore` arrays, which now gets called in more occasions (and thus a few existing tests).

Reply:
Would our existing [`normalize_dtype`](https://cs.github.com/napari/napari/blob/67e677106cd309a1de1ae5bf5c1be961212b70e5/napari/utils/_dtype.py?q=repo%3Anapari%2Fnapari+normalize_dtype#L54) be a preferred solution here?